File: 234-palindrome-linked-list/palindrome-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 */
var isPalindrome = function(head) {
    let slow=head;
    let fast=head;
    while(fast!=null && fast.next!=null){
        fast=fast.next.next;
        slow=slow.next;
    }
    let prev=null;
    while(slow!=null){
        let next=slow.next;
        slow.next=prev;
        prev=slow;
        slow=next;
    }
    while(prev!=null){
        if(prev.val!=head.val){
            return false;
        }
        prev=prev.next;
        head=head.next;
    }
    return true;
};

File: 234-palindrome-linked-list/palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def isPalindrome(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: bool
        """
        s=head
        f=head
        while(f and f.next):
            f=f.next.next
            s=s.next
        
        prev=None
        while s:
            nxt=s.next
            s.next=prev
            prev=s
            s=nxt
        while(prev):
            if prev.val != head.val:
                return False
            head=head.next
            prev=prev.next
        return True
